### PR TITLE
feat: `avm/res/compute/image`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.compute.image](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.compute.image.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

**Comment by segraef** [comment]
> Requires uncommenting in the `/.github/CODEOWNERS` file

done

File: avm/res/compute/image/main.bicep [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+])
+param osState string = 'Generalized'
+
+@description('Optional. The snapshot resource ID.')
+param snapshotResourceId string = ''
+
+@description('Optional. Specifies the parameters that are used to add a data disk to a virtual machine.')
+param dataDisks array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
```
Comment (AlexanderSehr) [inline avm/res/compute/image/main.bicep:63]:
```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #/avm/res/compute/disk/ @Azure/avm-res-compute-disk-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/compute/disk-encryption-set/ @Azure/avm-res-compute-diskencryptionset-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/compute/gallery/ @Azure/avm-res-compute-gallery-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/compute/image/ @Azure/avm-res-compute-image-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/compute/image/ @Azure/avm-res-compute-image-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (eriqua) [inline .github/CODEOWNERS:28]: Please ref to https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

Comment (segraef) [reply .github/CODEOWNERS:28]: @Azure/avm-res-compute-image-module-owners-bicep
@Azure/avm-res-compute-image-module-contributors-bicep

File: avm/res/compute/image/tests/e2e/defaults/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
+param resourceGroupName string = 'dep-${namePrefix}-compute.images-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = 'eastus'
```
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/defaults/main.test.bicep:16]: Will be overwritten. Please refer to: https://github.com/Azure/bicep-registry-modules/pull/792#discussion_r1455072261

Comment (segraef) [reply avm/res/compute/image/tests/e2e/defaults/main.test.bicep:16]: done

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**